### PR TITLE
Add features to README

### LINKED ISSUE #92: Expose secret as env variable
**Describe the solution you'd like**
I would like to be able to sync Key Vault secrets and expose them as an environment variables as well. As far as I know it is only possible to expose them via a volume, but a lot of applications still use environment variables (even thought that maybe not best practice).

**Anything else you would like to add:**
No

**Environment:**

- Secrets Store CSI Driver version: (use the image tag): <latest>
- Azure Key Vault provider version: (use the image tag): <latest>
- Kubernetes version: (use `kubectl version`): <1.16.x>
- Cluster type: (e.g. AKS, aks-engine, etc): AKS

### PR DESCRIPTION
**Pull request by ritazh**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #92 

**Special notes for your reviewer**:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,81 +97,94 @@ Take note of the following properties for use in the next section:
...
-              objectVersion: ""         # [OPTIONAL] object versions, default to latest if empty
-            - |
-              objectName: key1
-              objectAlias: ""
-              objectType: key
-              objectVersion: ""
-        resourceGroup: "rg1"            # [REQUIRED for version < 0.0.4] the resource group of the KeyVault
-        subscriptionId: "subid"         # [REQUIRED for version < 0.0.4] the subscription ID of the KeyVault
-        tenantId: "tid"                 # the tenant ID of the KeyVault
+#### Create your own SecretProviderClass Object
```
Comment (bdlb77) [inline README.md:100]:
```suggestion
#### Create Your Own SecretProviderClass Object
```
nit: spelling capitalization for heading

**Comment by ritazh** [comment]
cc @phillipgibson FYI

File: README.md [hunk 2]
```diff
@@ -1,16 +1,36 @@
...
+- Supports windows containers (Kubernetes version v1.18+)
+- Supports sync with Kubernetes Secrets (Secrets Store CSI Driver v0.0.10+)
+- Supports multiple secrets stores providers in the same cluster.
+
+#### Table of Contents
+
+- [Demo](#demo)
+- [Usage](#usage)
+  - [Install with Helm](#install-the-secrets-store-csi-driver-and-the-azure-keyvault-provider)
+  - [Using the Azure Key Vault Provider](##using-the-azure-key-vault-provider)
```
Comment (aramase) [inline README.md:22]: nit: s/##using-the-azure-key-vault-provider/#using-the-azure-key-vault-provider

File: README.md [hunk 3]
```diff
@@ -1,16 +1,36 @@
...
+- [Azure Key Vault Provider Features](#azure-key-vault-provider-features)
+- [Troubleshooting](#troubleshooting)
+- [Contributing](#contributing)
```
Comment (aramase) [inline README.md:25]: Should we also add `Testing` and `Support` to the table of contents?

File: README.md [hunk 4]
```diff
@@ -76,105 +64,185 @@ Take note of the following properties for use in the next section:
...
+  | objects                | yes      | a string of arrays of strings                                   | ""            |
+  | objectName             | yes      | name of a Key Vault object                                      | ""            |
+  | objectAlias            | no       | [__*available for version > 0.0.4*__] specify the filename of the object when written to disk - defaults to objectName if not provided | "" |
+  | objectType             | yes      | type of a Key Vault object: secret, key or cert                 | ""            |
+  | objectVersion          | no       | version of a Key Vault object, if not provided, will use latest | ""            |
+  | resourceGroup          | no      | [__*required for version < 0.0.4*__] name of resource group containing key vault instance            | ""            |
+  | subscriptionId         | no      | [__*required for version < 0.0.4*__] subscription ID containing key vault instance                   | ""            |
+  | tenantId               | yes      | tenant ID containing key vault instance                         | ""            |
+
+#### Provide Identity to Access Key Vault
```
Comment (aramase) [inline README.md:123]: This step is referenced in `Create your own SecretProviderClass Object`. Instead would it be better to make this the second step, right after `Create a new Azure Key Vault resource or use an existing one`?

File: README.md [hunk 5]
```diff
@@ -76,105 +64,185 @@ Take note of the following properties for use in the next section:
...
-              volumeAttributes:
-                secretProviderClass: "azure-kvname"
-      ```
-
-  1. Select and complete an option from [below to enable access to the Key Vault](#provide-identity-to-access-key-vault)
-
-  1. Deploy the secretProviderClass
+#### Create your own SecretProviderClass Object
+
+Create a `secretproviderclasses` custom resource to provide provider-specific parameters for the Secrets Store CSI driver. In this example, use an existing Azure Key Vault or the Azure Key Vault resource created previously.
```
Comment (aramase) [inline README.md:69]: Can we use `SecretProviderClass` instead of `secretproviderclasses`?

File: README.md [hunk 6]
```diff
@@ -76,105 +64,185 @@ Take note of the following properties for use in the next section:
...
+
+Update [this sample deployment](examples/v1alpha1_secretproviderclass.yaml) to create a `secretproviderclasses` resource to provide Azure-specific parameters for the Secrets Store CSI driver.
```
Comment (aramase) [inline README.md:71]: ditto

File: README.md [hunk 7]
```diff
@@ -76,105 +64,185 @@ Take note of the following properties for use in the next section:
...
+1. [Service Principal](docs/service-principal-mode.md)
+1. [Pod Identity](docs/pod-identity-mode.md)
+1. [VMSS User Assigned Managed Identity](docs/user-assigned-msi-mode.md)
+1. [VMSS System Assigned Managed Identity](docs/system-assigned-msi-mode.md)
+
+#### Update your Deployment Yaml
+
+To ensure your application is using the Secrets Store CSI driver, update your deployment yaml to use the `secrets-store.csi.k8s.io` driver and reference the `SecretProviderClass` resource created in the previous step.
+
+Update your [linux deployment yaml](examples/nginx-pod-secrets-store-inline-volume-secretproviderclass.yaml) or [windows deployment yaml](examples/windows-pod-secrets-store-inline-volume-secret-providerclass.yaml) to use the Secrets Store CSI driver and reference the `secretProviderClass` resource created in the previous step.
```
Comment (aramase) [inline README.md:136]: s/secretProviderClass/SecretProviderClass

File: docs/install-yamls.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+For windows nodes
+```bash
+kubectl apply -f https://raw.githubusercontent.com/Azure/secrets-store-csi-driver-provider-azure/master/deployment/provider-azure-installer-windows.yaml
+```
+
+To validate the provider's installer is running as expected, run the following commands:
+
+```bash
+kubectl get pods -l app=csi-secrets-store-provider-azure
```
Comment (aramase) [inline docs/install-yamls.md:29]: do we also want to add `kubectl get pods -l app=csi-secrets-store` to validate driver pods are up and running?

File: README.md (hunk 3 above)
Comment (ritazh) [reply README.md:25]: Good call.

File: README.md (hunk 4 above)
Comment (ritazh) [reply README.md:123]: All 4 modes reference updates to the `SecretProviderClass` resource or the application deployment yaml, which would make more sense _after_ the user has gone thru the `Create your own SecretProviderClass Object` section. WDYT?

Comment (aramase) [reply README.md:125]: Yeah, makes sense!

**Pull request merged**